common/client-core/src/client/topology_control/geo_aware_provider.rs

```diff
@@ -0,0 +1,263 @@
+use std::{collections::HashMap, fmt};
+
+use log::{debug, error, info};
+use nym_explorer_api_requests::PrettyDetailedMixNodeBond;
+use nym_topology::{
+    mix::Layer,
+    nym_topology_from_detailed,
+    provider_trait::{async_trait, TopologyProvider},
+    NymTopology,
+};
+use nym_validator_client::client::{MixId, MixNodeDetails};
+use url::Url;
+
+const EXPLORER_API_MIXNODES_URL: &str = "https://explorer.nymtech.net/api/v1/mix-nodes";
+
+async fn fetch_mixnodes_from_explorer_api() -> Option<Vec<PrettyDetailedMixNodeBond>> {
```

Comment:
i actually dont remember, but dont we have an explorer api client?

Reply:
I didn't find one :(